velox/exec/Window.cpp

```diff
@@ -374,19 +346,51 @@ void Window::callApplyForPartitionRows(
     // as WindowFunction only sees one partition at a time.
     rawPeerStarts[j] = peerStartRow_ - firstPartitionRow;
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
+  }
 
-    for (auto w = 0; w < numFuncs; w++) {
-      auto frameEndPoints = findFrameEndPoints(
-          w,
-          firstPartitionRow,
-          lastPartitionRow,
-          peerStartRow_,
-          peerEndRow_ - 1,
-          i);
-      rawFrameStartBuffers[w][j] = frameEndPoints.first - firstPartitionRow;
-      rawFrameEndBuffers[w][j] = frameEndPoints.second - firstPartitionRow;
+  auto updateFrameBounds = [&](vector_size_t* frameBoundBuffer,
+                               core::WindowNode::BoundType boundType,
+                               core::WindowNode::WindowType type,
+                               bool isStartBound) -> void {
+    switch (boundType) {
+      case core::WindowNode::BoundType::kUnboundedPreceding:
+        std::memset(frameBoundBuffer, 0, numRows * sizeof(vector_size_t));
+        break;
+      case core::WindowNode::BoundType::kUnboundedFollowing:
+        std::fill_n(
+            frameBoundBuffer, numRows, lastPartitionRow - firstPartitionRow);
+        break;
+      case core::WindowNode::BoundType::kCurrentRow: {
+        if (type == core::WindowNode::WindowType::kRange) {
+          vector_size_t* rawPeerBuffer =
+              isStartBound ? rawPeerStarts : rawPeerEnds;
+          std::copy(rawPeerBuffer, rawPeerBuffer + numRows, frameBoundBuffer);
+        } else {
+          VELOX_NYI("Not supported");
+        }
+        break;
+      }
+      case core::WindowNode::BoundType::kPreceding:
+      case core::WindowNode::BoundType::kFollowing:
+        VELOX_NYI("Not supported");
+      default:
+        VELOX_USER_FAIL("Invalid frame start value");
```

Comment:
Is this message accurate? "start" doesn't apply when isStartBound is false.

Reply:
Apologies for missing this, I have updated it to "Invalid frame bound type".